src/scenic/core/regions.py

```diff
@@ -2055,6 +2062,152 @@ def evaluateInner(self, context):
         )
 
 
+class VoxelRegion(Region):
+    """Region represented by a voxel grid in 3D space.
+
+    Args:
+        encoding: A numpy array encoding the voxel grid.
+        dimensions: An optional 3-tuple, with the values representing width, length, height
+          respectively. The voxel region will be scaled to have these dimensions. Default
+          value (1,1,1).
+        position: A position, which determines where the center of the region will be. Default
+          value is the origin, (0,0,0).
+        voxelGrid: Optionally, just pass in the final trimesh voxelGrid to be used. Passing this
+          parameter will result in encoding, dimensions, and position being ignored.
+        orientation: An optional vector field describing the preferred orientation at every point in
+          the region.
+        name: An optional name to help with debugging.
+        lazy: Whether or not to be lazy about pre-computing internal values. Set this to True if this
+          VoxelRegion is unlikely to be used outside of an intermediate step in compiling/pruning.
+    """
+
+    def __init__(self, voxelGrid, orientation=None, name=None, lazy=False):
+        # Initialize superclass
+        super().__init__(name, orientation=orientation)
+
+        # Work around Trimesh caching bug
+        self._voxelGrid = trimesh.voxel.VoxelGrid(
+            voxelGrid.encoding, transform=voxelGrid.transform.copy()
+        )
+
+        # Check that the encoding isn't empty. In that case, raise an error.
+        if self._voxelGrid.encoding.is_empty:
+            raise ValueError("Tried to create an empty VoxelRegion.")
+
+        # Transform voxel grid points and extract scale
+        self.voxel_points = self._voxelGrid.points
+        self.scale = self._voxelGrid.scale
+
+        # Initialize KD-Tree for containment checking if not lazy
+        if not lazy:
+            self.kdTree
```

Comment:
What's the rationale for forcing the computation here rather than just having it be computed the first time it's needed? (In other words, why provide `lazy` as an option rather than just having it always be true?)

Reply:
This is to force computation at compile time instead of sample time (unless we expect this object not to need the kdTree, i.e. `lazy=True`).

Reply:
Removing as discussed offline.